### PR TITLE
client-go: deprecate TPR example and add CRD example

### PR DESCRIPTION
**Pull request by sttts**
/cc @nilebox

Part of https://github.com/kubernetes/kubernetes/issues/46702

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/client/cr.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	"k8s.io/apimachinery/pkg/util/wait"
+	apiv1 "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/rest"
+	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
+	// _ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const exampleCRDName = crv1.ExampleResourcePlural + "." + crv1.GroupName
+
+func CreateCustomResource(clientset apiextensionsclient.Interface) (*apiextensionsv1beta1.CustomResourceDefinition, error) {
```
Comment (deads2k) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/client/cr.go]: CreateCustomResourceDefinition.

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/main.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	controller := examplecontroller.ExampleController{
+		ExampleClient: exampleClient,
+		ExampleScheme: exampleScheme,
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	go controller.Run(ctx)
+
+	// Create an instance of our TPR
```
Comment (deads2k) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/main.go]: of our custom resource

**Comment by deads2k** [comment]
minor comments, lgtm otherwise.

**Comment by nilebox** [comment]
👍 actually was going to propose to make an example of CRD client instead of TPR one :)

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/apis/cr/v1/register.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var (
+	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
+	AddToScheme   = SchemeBuilder.AddToScheme
+)
+
+// GroupName is the group name use in this package.
```
Comment (nilebox) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/apis/cr/v1/register.go]: typo: use -> "used" or "to be used"

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/client/cr.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+			case apiextensionsv1beta1.NamesAccepted:
+				if cond.Status == apiextensionsv1beta1.ConditionFalse {
+					fmt.Printf("Name conflict: %v\n", cond.Reason)
+				}
+			}
+		}
+		return false, err
+	})
+	if err != nil {
+		clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(exampleCRDName, nil)
```
Comment (nilebox) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/client/cr.go]: handle error from delete request?

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/main.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+	}
+	defer apiextensionsclientset.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(crd.Name, nil)
+
+	// make a new config for our extension's API group, using the first config as a baseline
+	exampleClient, exampleScheme, err := exampleclient.NewClient(config)
+	if err != nil {
+		panic(err)
+	}
+
+	// start a controller on instances of our TPR
```
Comment (nilebox) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/main.go]: of our custom resource

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/client/cr.go (hunk 4 above)
Comment (nilebox) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/client/cr.go]: handle delete error

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/client/cr.go (hunk 1 above)
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/client/cr.go]: fixed

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/main.go (hunk 2 above)
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/main.go]: fixed

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/apis/cr/v1/register.go (hunk 3 above)
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/apis/cr/v1/register.go]: fixed

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/main.go (hunk 5 above)
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/main.go]: fixed

File: staging/src/k8s.io/apiextensions-apiserver/examples/client-go/client/cr.go (hunk 4 above)
Comment (sttts) [inline staging/src/k8s.io/apiextensions-apiserver/examples/client-go/client/cr.go]: fixed

**Comment by sttts** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Pull request merged**